installer/mod.ts

```diff
@@ -86,69 +87,104 @@ function createDirIfNotExists(path: string): void {
 
-  return paths.includes(path);
-}
+  const paths = envPath.split(isWindows ? ";" : ":");
 
-function getInstallerDir(): string {
-  const { HOME } = env();
+  let fileAbsolutePath = filePath;
 
-  if (!HOME) {
-    throw new Error("$HOME is not defined.");
-  }
-
-  return path.join(HOME, ".deno", "bin");
-}
-
-// TODO: fetch doesn't handle redirects yet - once it does this function
-//  can be removed
-async function fetchWithRedirects(
-  url: string,
-  redirectLimit: number = 10
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-): Promise<any> {
-  // TODO: `Response` is not exposed in global so 'any'
-  const response = await fetch(url);
-
-  if (response.status === 301 || response.status === 302) {
-    if (redirectLimit > 0) {
-      const redirectUrl = response.headers.get("location")!;
-      return await fetchWithRedirects(redirectUrl, redirectLimit - 1);
+  for (const p of paths) {
+    let pathInEnv = path.normalize(p);
+    // In Windows. We can get the path from env. eg. C:\Users\username\.deno\bin
+    // But in the path of Deno, there is no drive letter. eg \Users\username\.deno\bin in deno
+    if (isWindows) {
+      const driverLetterReg = /^[c-z]:/i;
+      if (driverLetterReg.test(pathInEnv)) {
+        fileAbsolutePath = HOMEDRIVE + "\\" + fileAbsolutePath;
+      }
+    }
+    if (pathInEnv === fileAbsolutePath) {
+      return true;
     }
+    fileAbsolutePath = filePath;
   }
 
-  return response;
+  return false;
 }
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-async function fetchModule(url: string): Promise<any> {
-  const response = await fetchWithRedirects(url);
+function getInstallerDir(): string {
+  // In Windows's Powershell. $HOME environmental variable maybe null. use $HOMEPATH instead.
+  let { HOME, HOMEPATH } = env();
+
+  const HOME_PATH = HOME || HOMEPATH;
 
-  if (response.status !== 200) {
-    // TODO: show more debug information like status and maybe body
-    throw new Error(`Failed to get remote script ${url}.`);
+  if (!HOME_PATH) {
+    throw new Error("$HOME is not defined.");
   }
 
-  const body = await readAll(response.body);
-  return decoder.decode(body);
+  return path.join(HOME_PATH, ".deno", "bin");
 }
 
 function showHelp(): void {
   console.log(`deno installer
   Install remote or local script as executables.
-  
+
 USAGE:
-  deno https://deno.land/std/installer/mod.ts EXE_NAME SCRIPT_URL [FLAGS...]  
+  deno https://deno.land/std/installer/mod.ts EXE_NAME SCRIPT_URL [FLAGS...]
 
 ARGS:
-  EXE_NAME  Name for executable     
+  EXE_NAME  Name for executable
   SCRIPT_URL  Local or remote URL of script to install
   [FLAGS...]  List of flags for script, both Deno permission and script specific flag can be used.
-  `);
+`);
+}
+
+async function generateExecutable(
+  filePath: string,
+  commands: string[]
+): Promise<void> {
+  // In Windows. if use Powershell to run the installed module. It will look up .cmd file.
```

Comment:
Replace with:
// On Windows if user is using Powershell `.cmd` extension is need to run the installed module. Generate batch script to satisfy that